### ISSUE TITLE
Signal should have background permission.

### ISSUE THREAD

**Issue by metal3d** [flathub/org.signal.Signal#177]
See https://github.com/signalapp/Signal-Desktop/issues/5120
By using the "--use-tray-icon" option (that should be the default IMHO), Signal stops after a few seconds.
The background permission should be added please.
Details in the original issue.

Thanks 😊

**Comment by bermeitinger-b** [comment]
We don't add our own defaults because the upstream packages also does not have this tray option as default. (see #152)

How can we add this special permission? I did not find anything about this and none of the other flatpaks (I looked at msteams, skype, riot, discord, etc.).

I'm starting it with `flatpak run org.signal.Signal --use-tray-icon --start-in-tray` and it just stays in the tray until I open it without quitting. How can I reproduce this?

**Comment by metal3d** [comment]
If I close the window, Signal is stopped after a few seconds because the backgorund permission is not set.

I had to type:
```
flatpak permission-set background background org.signal.Signal yes
```

This permission came with Gnome 3.36 (if I'm not wrong), and the application should ask this.

What I know, for now, it's that some application (like Jami) ask permission to run in background when I closed them the first time.  That's not the case with Signal application.

This is relative to flatpak (this permission is only asked with application that are packed with flatpak). I don't know how to help more, I never tried to build a flatpak app, but I will take a look.

I looked on others apps that have this permission (that was asked to me), Bitwarden for example.

https://github.com/flathub/com.bitwarden.desktop

I don't find anything relevent, but maybe the "--tak-name", or "--ipc" options has something to deal with it ?

I can shwo you the permissions I have, the only app that needed to set permission in command line is Signal:
```
flatpak permission-list 
Table         Object       App                           Permissions                  Data
gnome         geolocation  org.gnome.Calendar            EXACT,1541900128             []
gnome         geolocation  org.gnome.Weather.Application CITY,1542782857              []
documents     392b37c2     org.telegram.desktop          read,write,grant-permissions (b'/home/metal3d/Images/del-phine/39891_427343433855_3418890_n.jpg', 54, 17896394, 0)
documents     8c433a63     org.telegram.desktop          read,write,grant-permissions (b'/home/metal3d/Images/bab/738066_4601884978917_545666540_o.jpg', 54, 17896315, 0)
notifications notification com.uploadedlobster.peek      yes                          0x00
notifications notification org.nicotine_plus.Nicotine    yes                          0x00
background    background   com.bitwarden.desktop         yes                          0x00
background    background   com.google.AndroidStudio      no                           0x00
background    background   com.uploadedlobster.peek      yes                          0x00
background    background   com.valvesoftware.Steam       no                           0x00
background    background   net.jami.Jami                 yes                          0x00
background    background   org.gnome.FeedReader          no                           0x00
background    background   org.signal.Signal             yes                          0x00
background    background   org.telegram.desktop          yes                          0x00
```

**Comment by metal3d** [comment]
Oh... maybe...
```
"runtime": "org.freedesktop.Platform",
"runtime-version": "20.08",
```

The application that have the background permission here are based on `org.gnome.Platform` runtime. Maybe...

**Comment by metal3d** [comment]
Found this:
https://docs.flatpak.org/en/latest/desktop-integration.html

Following freedesktop integration:
https://flatpak.github.io/xdg-desktop-portal/portal-docs.html#gdbus-org.freedesktop.portal.Background

**Comment by metal3d** [comment]
I precise, because I was probably not clear.

- Without permission, Signal stops when I close the window
- With permission, Signal stays in tray

**Comment by metal3d** [comment]
So, Reading: https://github.com/flatpak/xdg-desktop-portal, it seems that adding `--talk-name=org.freedesktop.portal.Background` should do the job ?

https://github.com/flatpak/xdg-desktop-portal/blob/master/data/org.freedesktop.portal.Background.xml

**Comment by bermeitinger-b** [comment]
Can you test the build in #180 ? 

As I can't reproduce, I'm flying blind here. You could test this locally by building it yourself.

**Comment by metal3d** [comment]
I will try as soon as possible (max tomorrow) - I'm currently working on
something.
That's weird you don't have the problem, can you tell me the gnome version
you're using ?

**Comment by bermeitinger-b** [comment]
I'm not using Gnome.

**Comment by metal3d** [comment]
That makes sense you don't have the problem so 😊

**Comment by bermeitinger-b** [comment]
This should now be fixed by #180 . At least you should be asked correctly if Signal is allowed to run in the background.
